### ISSUE TITLE
Websocket subscription request: Need clarification

### ISSUE THREAD

**Issue by jasperpeeters** [AMWA-TV/nmos-discovery-registration#80]
I need some clarification regarding secured websocket subscriptions.

Since version IS04-v1.1 a `secure` flag exists in the schema of the `POST` request to create such a subscription:

```
    "secure": {
      "description": "Whether to produce a secure websocket connection (wss://). NB: Default should be 'false' if the API is being presented via HTTP, and 'true' for HTTPS",
      "type": "boolean"
    },
```
From: [queryapi-subscriptions-post-request.json](https://amwa-tv.github.io/nmos-discovery-registration/tags/v1.2.1/html-APIs/schemas/queryapi-subscriptions-post-request.json)

If I run a secured registry, the normal case would be that I don't want to allow unsecured subscriptions. I assume that I should respond with a `400`, but this case is not that clear as for example in IS05 where we have constraints which I can check in advance as a controller. If that is correct, maybe the docs can be updated to explicitly mention this. Current text only says that a `400` must be `returned when the POST request is incorrectly formatted or missing mandatory attributes`.

In general: the flag `secure` is meant to be used for registries with interfaces in different subnets (e.g. one local which can be unsecured, and one external which has to be secured)? I currently cannot make up any other scenario where running an unsecured registry and allowing secured websocket connections or running a secured registry and allowing unsecured websocket connections would be required.

**Comment by andrewbonney** [comment]
I agree that some extra docs are required here. We have the same situation in our implementation and issue a 400 if an insecure websocket is requested via an https connection.

I think in hindsight the 'secure' flag here is probably unnecessary and can be inferred from whether the registry is accessed via http/https. The only use I've seen from it has been in prototyping where a registry is accessible via both http and https (I guess a similar scenario to the one you describe), and you need to switch the flag to its different states.

**Comment by garethsb** [comment]
I had the same question. If I'm talking to a Query API via HTTPS, why would I want the WebSocket not to be WSS. And vice-versa, it would seem very odd to be talkign via HTTP and want a WSS connection.

**Comment by andrewbonney** [comment]
This is being addressed as part of https://github.com/AMWA-TV/nmos-discovery-registration/pull/84

**Comment by garethsb** [comment]
I.e. by https://github.com/AMWA-TV/nmos-discovery-registration/blob/8c17a74c9592b947ae7cbcc8fdb95b4019a3d424/docs/4.2.%20Behaviour%20-%20Querying.md#subscriptions

**Comment by jasperpeeters** [comment]
> This is being addressed as part of #84

Much clearer now :+1:
